.github/workflows/release-pr.yml

```diff
@@ -0,0 +1,64 @@
+name: Changelog
+on:
+  push:
+    branches:
+      - release-*
+
+jobs:
+  release-notes-pr:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Extract branch
+        id: extract_branch
+        run: echo "branch=${GITHUB_REF#refs/heads/}" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_REF: ${{ github.ref_name }}
+
+      - name: Extract tag from branch name
+        id: extract_tag
+        run: echo "tag=${BRANCH#release-}" >> "$GITHUB_OUTPUT"
+        env:
+          BRANCH: ${{ steps.extract_branch.outputs.branch }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Get latest release
+        id: get_latest_release
+        run: |
+          latest_release=$(gh release view --json tagName --jq .tagName)
+          echo $latest_release
+          echo "tag=$latest_release" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate release notes
+        id: generate_release_notes
+        # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release
+        run: |
+          response=$(curl -L \
```

Comment:
What if this curl command will fail from some rate-limiting rule? How to handle such errors?

Reply:
This hasn't failed for me in my tests. I think for now we can go forward by just retrying the action manually if it fails and addressing it in a fix PR. I don't think we should complicate it now :)